### ISSUE TITLE
.lfsconfig deleted after checking out old commits after git lfs migrate

### ISSUE THREAD

**Issue by jomark-t** [git-lfs/git-lfs#3046]
Hi again,

I migrated our repository to git lfs, everything went smooth until one developer tried checking out older commits. I fails because .lfsconfig will be deleted and fails to git lfs fetch/pull the files. Are there any workaround?

Much thanks!

**Comment by jomark-t** [comment]
I failed to migrate another branch. So we are now trying to do diffs/patch and there are 150+ patches :( I was hoping if there's a command to add that specific branch without re-migrating all(running git lfs migrate again). Fortunately we have a backup of our repository.

**Comment by ttaylorr** [comment]
Hi @jomark-t! Thanks for opening this issue. I'm not quite sure on what's going wrong immediately: what was in the .lfsconfig file that is now missing? The `git lfs migrate` tool does not remove any files, so I am curious to see what command you are running.

`git lfs migrate import` also allows for specific branches to be imported, i.e., with `git lfs migrate import [<ref [...]>]`, like, `git lfs migrate import my-feature`.

**Comment by jomark-t** [comment]
Hi @ttaylorr,

I'm working on this migration from SVN to GIT for months now, I have successfully migrated it using Gerrit as a version control host. 

Next step, storing binary files to a storage server, we use JFrog Artifactory. I've tested it, all is well.

Next step, its time to migrate it. 

1. Add `.lfsconfig` to repository (master and official branches), committed and pushed.
2. Run `git lfs migrate`
3. Checked if all binaries are stored in the storage server. Cloning has no issues.

The next day,  one of the devs from other site complained that their branch is gone. I've only migrated official branches and master. Due to miscommunication their branch is gone. Although I have the repository backed up before migrating.

They tried checking out certain point in the history and it fails. 

```
$ git checkout -b LTE4042_GROUP_F_NEW da3f81664c305c3103e01be47595f8891b8f56e8
Downloading confidential_path/confidential_file.tar.gz (860 KB)
Error downloading object: confidential_path/confidential_file.tar.gz (2b3527f): Smudge error: Error downloading confidential_path/confidential_file.tar.gz (2b3527f2c93806919ec1be39e524efc8889db1667e47ffaea8004e8ae40783e3): batch request: fatal: Gerrit Code Review: git-lfs-authenticate: not found: exit status 1

Errors logged to D:\<path>\.git\lfs\objects\logs\20180605T150232.2668065.log
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: confidential_path/confidential_file.tar.gz: smudge filter lfs failed

```
It seems that the .lfsconfig is gone? I'm not sure.

How to add their branch to newly migrated repository?

**Comment by ttaylorr** [comment]
> It seems that the .lfsconfig is gone? I'm not sure.

Ah, I think I understand the issue now. The `git lfs migrate` command will not "bring back" any extant `.lfsconfig` file. I.e., if you have a `.lfsconfig` file on the tip of `master` before migrating that includes configuration details that are necessary in order to checkout, and you migrate the branch `master`, `git lfs migrate` will not backport the changes from `.lfsconfig` to earlier points in history.

In other words, you will be unable to checkout those points in history prior to `.lfsconfig`, independent of `git lfs migrate`.

I don't think that this is a feature that we will add to `git lfs migrate import`. If you desire this kind of history modification, it should be sufficiently achieved by `git filter-branch`.
